drivers/modem/gsm_ppp.c

```diff
@@ -716,7 +718,7 @@ void gsm_ppp_start(const struct device *dev)
 
 	/* Re-init underlying UART comms */
 	int r = modem_iface_uart_init_dev(&gsm->context.iface,
-					  CONFIG_MODEM_GSM_UART_NAME);
+					  GSM_UART_DEV_NAME);
```

Comment:
It looks like this is a modem-specific private API that is only used by this modem driver, so it can be safely changed to take a `const struct device*` instead of a string name.

Reply:
I think this and the following one should be made into another PR, as the following ones involved other modems and even ESP drivers and this PR is supposed to convert the gsm_ppp to use dts only.